core/corehttp/gateway.go

```diff
@@ -132,7 +132,11 @@ func newGatewayBackend(n *core.IpfsNode) (gateway.IPFSBackend, error) {
 		}
 	}
 
-	backend, err := gateway.NewBlocksBackend(bserv, gateway.WithValueStore(vsRouting), gateway.WithNameSystem(nsys))
+	backend, err := gateway.NewBlocksBackend(bserv,
+		gateway.WithValueStore(vsRouting),
+		gateway.WithNameSystem(nsys),
+		gateway.WithResolver(n.UnixFSPathResolver),
```

Comment:
@hsanjuan  hm.. mind  we have offline exchange in two places ([here](https://github.com/ipfs/kubo/blob/89a476948b1d57cff3c7f07457aaccf2ea4feaf9/core/corehttp/gateway.go#L115) and [here](https://github.com/ipfs/kubo/blob/89a476948b1d57cff3c7f07457aaccf2ea4feaf9/core/corehttp/gateway.go#L82)). 

If I understand correctly, by using global path resolver here, that uses different (unlimited) exchange backend, we ignore the gateway-specific no-fetch preference and path resolution may cause raw blocks to be fetched from the network when it did not happen before 

If the content fits in a single final block, gateway will host third-party content  when user explicitly asked to not fetch third-party data.

Reply:
in the libp2pGatewayOption I don't see an issue.

In the other one I see it... we should create the resolver using the fetcher that uses the offline blockservice. I'll make that change.

Reply:
see b4b7ca71d